plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -388,16 +398,25 @@ function ListMenuItem:update()
 
             -- Build the right widget
 
-            local wfileinfo = TextWidget:new{
-                text = fileinfo_str,
-                face = Font:getFace("cfont", 14),
-                fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
-            }
-            local wpageinfo = TextWidget:new{
-                text = pages_str,
-                face = Font:getFace("cfont", 14),
-                fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
-            }
+            local fontsize_info = 16
+            local wfileinfo
+            local wpageinfo
+            -- Build file and page info texts with decreasing font size
+            -- till it fits in the space available
+            repeat
+                wfileinfo = TextWidget:new{
```

Comment:
For the right stuff, where file size and type is always one line, and page info another line, I don't think you need to do this repeat stuff and build a widget each time and decrase font size. (It's uneeded cpu/power usage/latency - but it is ok/needed when we have vertical space where we can fit a variiable amount of lines and as much as the text - which is not the case here.)
You ought to be able to compute the right font size that will fit - or just find some value from a factor of the available height.

Reply:
Good point - I followed the example from the title and author, where the text length may vary.
Do you have an idea on how to compute the text widget height from the font size? Otherwise, I'll have to dig into documentation and code, I don't want to estimate.

Reply:
I'd say, if previously we had:
```lua
item_height_min = Screen:scaleBySize(64)
face = Font:getFace("cfont", 14) -- 14 gets scaleBySize'd too
```
A factor of `item_height * 14 / 64` might be worth trying. (eventually adjust it to 16/64 = 1/4 :)

Otherwise, there's that:
https://github.com/koreader/koreader/blob/a71b6250e2ba25476fb4f09b1d1a1b24d6275bd4/frontend/ui/widget/textwidget.lua#L55-L65